osu.Game/Screens/Select/Carousel/CarouselBeatmapSet.cs

```diff
@@ -132,6 +124,14 @@ public override int CompareTo(FilterCriteria criteria, CarouselItem other)
             return otherSet.BeatmapSet.ID.CompareTo(BeatmapSet.ID);
         }
 
+        private static int compareNullableDateTimeOffset(DateTimeOffset? x, DateTimeOffset? y)
+        {
+            if (x == null) return 1;
+            if (y == null) return -1;
+
+            return y.Value.CompareTo(x.Value);
+        }
```

Comment:
This comparator is incorrect, as `compareNullableDateTimeOffset(null, null)` will yield 1.

There is actually a method to do this in the standard library - see [`Nullable.Compare<T>(Nullable<T>, Nullable<T>)`](https://learn.microsoft.com/en-us/dotnet/api/system.nullable.compare?view=net-7.0).

As far as I can tell if the built-in method is to be used, it will have to be called with the operands swapped, because this method is also swapping them (as in, `Nullable.Compare(otherSet.BeatmapSet.DateRanked, BeatmapSet.DateRanked)` and so on). It passes tests, so I assume it will behave correctly, but this method is kind of confusing me what with the `x` and `y` comparison inversion.

Reply:
Nice find.